### DISCUSSION TITLE
MQTTv5 state

### DISCUSSION THREAD

**Discussion by voxelias** [bytebeamio/rumqtt#663] [Q&A] [answered]
Hi!

What is the current state off the v5 implementation? The checkbox in the repo's readme is not marked, but I see the `v5` rust module and it seems functional though.

Thank you in Advance!

**Comment by swanandx** [comment, marked as answer]
Hey @Lowentwickler , if you are talking about `rumqttc`, we do support [several](https://bytebeam.io/blog/rumqttc-now-supports-mqtt5-3/) v5 features. I think it isn't marked because it hasn't been much battle tested yet or maybe not cause some small v5 features are pending ( not really sure ).

For rumqttd, we are actively working on it, you can refer to #621 for tracking progress :100:. Planning to support all major features  within next few months.

**Reply by voxelias** [reply]
Hey @swanandx ! Thank you for response! Yes, I am interested in rumqttc features.

**Reply by swanandx** [reply]
Cool, we do fully support MQTTv5 in rumqttc!

**Reply by swanandx** [reply]
we can mark the checkbox in README :100: have u found any issues in functionality? if not, we can go ahead and mark in in readme

**Reply by voxelias** [reply]
Yesterday I've been testing response-request feature and it seems functional. So LGTM.  :)

**Reply by swanandx** [reply]
Nice, I have marked the checkbox in README, thanks for pointing it out :rocket:
